source/AI.cpp

```diff
@@ -3385,35 +3385,128 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 	{
-		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
-		// `Shift` results in boarding only player-owned escorts in need of assistance.
+		// Depending on the preference, board the (nearest/most expensive/mix of both) disabled ship.
+		// Pressing the board key multiple times cycles between eligible targets.
+		// It will be focusing on hostiles before allies. Holding `Shift`
+		// results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target)
+				|| (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			const auto boardingPriority = Preferences::GetBoardingPriority();
+
+			auto strategy = [&]() noexcept -> function<double(Ship &)>
+			{
+				Point current = ship.Position();
+				switch(boardingPriority)
 				{
-					if(shift && !other->IsYours())
-						continue;
+					case Preferences::BoardingPriority::value:
+						return [this, &ship](Ship &other) noexcept -> double
+						{
+							// Use the exact cost if the ship was scanned, otherwise use an estimation.
+							return this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+						};
+					case Preferences::BoardingPriority::mixed:
+						return [this, &ship, current](Ship &other) noexcept -> double
+						{
+							double cost = this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+							// Even if we divide by 0, doubles can contain and handle infinity,
+							// and we should definitely board that one then.
+							return cost * cost / (current.DistanceSquared(other.Position()) + 0.1);
+						};
+					// Default to distance priorities (the default setting).
+					default:
+					case Preferences::BoardingPriority::proximity:
+						return [current](Ship &other) noexcept -> double
+						{
+							return current.DistanceSquared(other.Position());
+						};
+				}
+			}();
+
+			using shipValue = pair<Ship *, double>;
+			auto boardable = vector<shipValue>{};
+
+			auto fillBoardable = [&ship, &foundEnemy, &boardable, &strategy](Ship &other) noexcept
+			{
+				if(CanBoard(ship, other))
+					boardable.emplace_back(&other, strategy(other));
+			};
+
+			if(shift)
+			{
+				for(Ship *escort : governmentRosters[ship.GetGovernment()])
+					if(escort->GetSystem() == ship.GetSystem())
+						fillBoardable(*escort);
+			}
+			else
+			{
+				auto ships = GetShipsList(ship, true);
+				boardable.reserve(ships.capacity());
```

Comment:
```suggestion
				boardable.reserve(ships.size());
```

Reply:
peter said to use capacity xD

Reply:
Peter was talking about `boardable.capacity()` and it is kinda important to understand why `boardable.capacity()` and `ships.size()` are relevant in this context, while `ships.capacity()` is not relevant.

A [short Google search](https://www.google.com/search?q=c%2B%2B+vector+reserve) usually results in nice tutorials and useful sites like [cplusplus.com](https://cplusplus.com/reference/vector/vector/reserve/) and [cppreference.com](https://en.cppreference.com/w/cpp/container/vector/reserve) that provide good background information on how sizes, capacities and reservations relate to each-other.

Reply:
Well .reserve without argument doesn't work so I got confused and tried to quickfix it

Reply:
Should be good now?